### PR TITLE
FailwithBadUsage: fix some false positive and add unit test

### PR DESCRIPTION
**Pull request by janus**
* FailwithBadUsage: new unit test that covers false positive

* FailwithBadUsage: fix false positive

* Fixes: https://github.com/fsprojects/FSharpLint/issues/539

### CODE REVIEW

File: src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs [hunk 1]
```diff
@@ -81,7 +86,7 @@ let private runner (args: AstNodeRuleParams) =
                     range
                     (BadUsageType.EmptyMessage)
                     (None)
-            | SynExpr.Null range -> 
+            | SynExpr.Null range ->
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs:89]: @janus this line change is not needed

File: src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs [hunk 2]
```diff
@@ -62,14 +63,18 @@ let private runner (args: AstNodeRuleParams) =
...
+
+                if testRange then
                     generateError failwithId.idText id range BadUsageType.DuplicateMessage maybeIdentifier
                 else
                     match maybeIdentifier with
                     | Some maybeId ->
                         generateError failwithId.idText id range BadUsageType.SwallowedException (Some maybeId)
                     | _ ->
-                        failwithErrorMessageList <- failwithErrorMessageList.Add(id)
+                        failwithRangeList <- List.append (List.singleton (id, range)) failwithRangeList
```
Comment (knocte) [inline src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs:77]: AFAIU you don't need to use append here, you can just concatenate with `::`

**Comment by knocte** [comment]
@aarani can you review please?

**Comment by aarani** [comment]
> @aarani can you review please?

I think by adding range to the list, you are just masking the underlying issue,
why there are multiple items (with the same range) in the list, in the first place.
If there are multiple list items for one failwith that shouldn't be the intended behavior and it should be fixed.

P.S: was it really necessary to change Set to List?

File: src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs (hunk 1 above)
Comment (janus) [reply src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs:89]: Done

File: src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs (hunk 2 above)
Comment (janus) [reply src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs:77]: Done

**Comment by knocte** [comment]
Given that @janus and @aarani had some additional discussion about this off-line (well, online, but outside GitHub), I'll update the state of things here:
* Emeka discovered that the reason for the failwith to be inserted twice in the Set is not because of some enumeration/seq problem but because the AST for the testcase shows the failwith twice.
* I then suggested that maybe Emeka's initial solution is not as bad as we thought (adding the range), however, the filename should be added too (range alone is not enough). @aarani still thinks that this is a hacky solution (please feel free to elaborate here Afshin)
* I wonder if the AST containing the failwith twice is an FCS bug? To clear out this theory, one could try the web fantomas tool to check the AST of the testcase (because fantomas updates FCS versions more frequently, I guess they are already using a newer version than FSharpLint).

File: src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs (hunk 2 above)
Comment (knocte) [reply src/FSharpLint.Core/Rules/Conventions/RaiseWithTooManyArguments/FailwithBadUsage.fs:77]: @janus you shouldn't say "Done" until you push, I don't see the addressing of this issue in the PR.

**Pull request merged**